### DISCUSSION TITLE
How to get Lua function name which calls C# method?

### DISCUSSION THREAD

**Discussion by Andreispitz** [neolithos/neolua#154] [Q&A]
Hello,

I have multiple lua functions which are assigned to the same c# method "MyC#Method".
I need to find which is the Lua function which calls the c# method.
How can I find it without sending the lua function name as parameter?

- `luaEnv[functionName1] = new Func<int,int>(MyC#Method);'
- 'luaEnv[functionName2] = new Func<int,int>(MyC#Method);'
- 'luaEnv[functionNameN] = new Func<int,int>(MyC#Method);`

`public int MyC#Method(int number)
{
    string myLuaFunction = ????; // any idea how can I find the Lua function name which calls this method ?
}`

Thanks,
Andrei

**Comment by neolithos** [comment]
The best way is to include the name/argument in a dynamic delegate.

```c#

    luaEnv[functionName1] = new Func<int,int>((a,b)=>MyC#Method("1", a, b));
    luaEnv[functionName2] = new Func<int,int>((a,b)=>MyC#Method("2", a, b));
    luaEnv[functionNameN] = new Func<int,int>((a,b)=>MyC#Method("3", a, b));
```
